src/main/java/io/cdap/plugin/gcp/dataplex/sink/DataplexBatchSink.java

```diff
@@ -199,6 +201,36 @@ public void transform(StructuredRecord input, Emitter<KeyValue<Object, Object>>
   public void onRunFinish(boolean succeeded, BatchSinkContext context) {
     if (this.config.getAssetType().equalsIgnoreCase(DataplexConstants.STORAGE_BUCKET_ASSET_TYPE)) {
       emitMetricsForStorageBucket(succeeded, context);
+      if (succeeded) {
```

Comment:
Can we add a comment describing why we need this logic and why it's only applicable to GCS assets